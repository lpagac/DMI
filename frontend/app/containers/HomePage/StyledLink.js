import styled from 'styled-components';
import { Link } from 'react-router-dom';

export default styled(Link)`
  background-color: #70e000;
  color: #343a40;
  border-radius: 7px;
  font-size: 30px;
  padding: 10px;
  text-align: center;
  text-decoration: none;
`;
