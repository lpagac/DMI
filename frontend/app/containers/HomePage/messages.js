/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.HomePage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'Check out our latest strings!',
  },
  subHeader: {
    id: `${scope}.subHeader`,
    defaultMessage: 'Limited time only... Add your own string!',
  },
});
