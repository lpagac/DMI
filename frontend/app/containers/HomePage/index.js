/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components';
import H1 from 'components/H1';
import H3 from 'components/H3';
import messages from './messages';

const Wrapper = styled.div`
  text-align: center;
  text-transform: capitalize;
  margin-top: 60px;
`;

export default function HomePage() {
  return (
    <Wrapper>
      <Helmet>
        <title>Home Page</title>
        <meta name="description" content="Strings app Home Page" />
      </Helmet>
      <H1>
        <FormattedMessage {...messages.header} />
      </H1>
      <H3>
        <FormattedMessage {...messages.subHeader} />
      </H3>
    </Wrapper>
  );
}
