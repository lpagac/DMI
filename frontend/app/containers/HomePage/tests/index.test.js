import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { ConnectedRouter } from 'connected-react-router';
import { createMemoryHistory } from 'history';
import configureStore from '../../../configureStore';

import HomePage from '../index';

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const history = createMemoryHistory();
    const store = configureStore({}, history);
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <ConnectedRouter history={history}>
          <IntlProvider locale="en">
            <HomePage />
          </IntlProvider>
        </ConnectedRouter>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });
});
