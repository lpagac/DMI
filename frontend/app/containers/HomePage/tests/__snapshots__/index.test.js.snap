// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="HomePage__Wrapper-i81cdn-0 dIOkee"
>
  <h1
    class="H1-tncdg6-0 kWlXoI"
  >
    <span>
      Check out our latest strings!
    </span>
  </h1>
  <h3
    class="H3-sc-1nwhpi5-0 dQPBtj"
  >
    <span>
      Limited time only... Add your own string!
    </span>
  </h3>
</div>
`;
